core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -252,6 +251,7 @@ struct mmu_partition {
 	xlat_tbl_t *l2_ta_tables;
 	unsigned int xlat_tables_used;
 	unsigned int asid;
+	uint64_t *user_l1_entries[NUM_BASE_TABLES][CFG_TEE_CORE_NB_CORE];
```

Comment:
Would it be possible to use indexes with the type `uint8_t` instead of pointers in order to save some memory?

Reply:
Saving memory was my concern for not using such a solution in the first place :)
Do you mean indexes inside `xlat_tbl_t *xlat_tables;` array? Sounds good.
Can you confirm that it was indeed the root cause for the performance degradation? For some reason adding `time` to `xtest` doesn't show the time statistics for me.

Reply:
> Saving memory was my concern for not using such a solution in the first place :)

Aha

> Do you mean indexes inside `xlat_tbl_t *xlat_tables;` array? Sounds good.

Yes :-)

> Can you confirm that it was indeed the root cause for the performance degradation? For some reason adding `time` to `xtest` doesn't show the time statistics for me.

Sorry, I haven't tested this yet.